HUDOC.js

```diff
@@ -0,0 +1,502 @@
+	return id[1];
+}
+
+// Adds the type of judgment at the end of the name
+function getTypeBit(doc, url) { // TODO: Switch to 'url' once we use the API instead
+	var description = scrapeMetaData(doc, "typedescription");
+	var language = scrapeMetaData(doc, "typedescription");
+
+	//Simpler logic for French that just takes over the type (unless it's a judgment)
+	// TODO: Add comprehensive logic like for English
+	if (url.includes("hudoc.echr.coe.int/fre#")) {
+		if (description.includes("Arrêt")) {
+			return "";
+		}
+		else {
+			return " (" + description.split("(")[0].toLowerCase() + ")";
+		}
+	}
+
+	if (description.includes("Judgment")) {
+		if (description.includes("Just Satisfaction") && !description.includes("Merits")) {
+			return " (just satisfaction)";
+		}
+		if (description.includes("Objection") || description.includes("Jurisdiction")) {
+			return " (preliminary objections)";
+		}
+		// TODO: Distinguish " (friendly settlement)" and " (striking out)"
+		// 				Requires a queryMetaData function
+		if (description.includes("Struck out")) return " (striking out)";
+		if (description.includes("Interpretation"))	return " (interpretation)";
+		if (description.includes("Revision")) return " (revision)";
+
+		return "";
+	}
+
+	if (description.includes("Decision")) return " (dec.)";
+	if (description.includes("Communicated")) return " (communicated)"; // TODO: Check if correct
+	if (description.includes("Revision")) return " (dec. on revision)"; // TODO: Check if correct
+	if (description.includes("Res-")) return " (resolution)"; // TODO: Check if correct, maybe different Zotero type?
+
+	return "";
+}
+
+// Downloads the legal summary in HUDOC
+function getLegalSummary(item, appno) {
+
+	var appnoString = appno.join(" AND appno:");
+
+	// Save the French version if user visited the French version of the judgment
+	var doctypeString = "";
+	if (item.language == "fre") {
+		doctypeString = "doctype:CLINF";
+	}
+	else {
+		doctypeString = "doctype:CLIN";
+	}
+
+	var summaryUrl = "https://hudoc.echr.coe.int/app/query/results?query=contentsitename=ECHR "
+						+ appnoString + " AND "
+						+ doctypeString + " AND "
+						+ "(kpdate=\"" + item.dateDecided + "\")"
+						+ "&select=itemid"
+						+ "&sort=&start=0&length=500";
+
+	// Request id of legal summary
+	ZU.doGet(summaryUrl, function (json) {
+		json = JSON.parse(json);
+
+		if (json.resultcount >= 1) {
+			var textUrl = "https://hudoc.echr.coe.int/app/conversion/docx/html/body?library=ECHR&id="
+						+ json.results[0].columns.itemid;
+
+			Zotero.debug("Getting text of legal summary at: " + textUrl);
+
+			// Request text of legal summary
+			ZU.doGet(textUrl, function (text) {
+				//Remove styles and span tags
+				text = text
+					.replace(/<style>[\s\S]*<\/style>/g, "")
+					.replace(/<\/?span[^>]*>/g,"")
+					.replace(/class\=\'s\S+\'/g, "");
+
+				item.notes.push({
+					title: "HUDOC Legal Summary",
+					note: "<h1> HUDOC Legal Summary </h1> </br>" + text
```

Comment:
You're already adding a title, so is the \<h1> necessary?

Reply:
I believe it is necessary. At least in my case, the note title immediately disappears, since Zotero seems to grab the first line of the note to use as title

Reply:
There's no `title` property of notes. That should just be removed. (This was an error in the translator documentation.)

`</br>` is also invalid HTML — it should be `<br/>` if anything. But if the text is a single paragraph, it can be wrapped in `<p>` `</p>` instead.

Reply:
Good to know, thanks! Fixed it now